testbed/coav-sim.sh

```diff
@@ -0,0 +1,162 @@
+AP_PX4=0
+AP_APM=1
+
+# ArduCopter Variables
+APM_CMD="arducopter"
+APM_TCP_PORT_1=5760
+APM_TCP_PORT_2=5762
+
+# PX4 Variables
+PX4_DIR=${PX4_DIR:-"~/px4/Firmware"}
+PX4_CMD="make posix_sitl_default jmavsim"
+PX4_UDP_PORT_1=14550
+PX4_UDP_PORT_2=14540
+
+# Collision Avoidance and Gazebo variables
+GZSITL_UDP_PORT=15556
+COAV_GCS_UDP_PORT=15557
+
+# Set autopilot
+AUTOPILOT=${AUTOPILOT:-AP_PX4}
+
+run_autopilot () {
+        # Run SITL Simulator
+    if (("$AUTOPILOT" == "$AP_PX4")); then
+        cd $PX4_DIR
+        $PX4_CMD &
+        SITLID=$!
+        cd - > /dev/null
+
+        # Wait for sitl
+        sleep 15
+    elif (("$AUTOPILOT" == "$AP_APM")); then
+        cd $SCRIPT_DIR # sitl must run in the same dir of "eeprom.bin"
+        $APM_CMD --model x &
+        SITLID=$!
+        cd - > /dev/null
+
+        # Wait for sitl
+        sleep 5
+    fi
+}
+
+run_gazebo () {
+    WORLD="simple_obstacle.sdf"
+
+    # Gazebo engine without GUI.
+    # The log can be played through `gazebo -p logfile`
+    SDFFILE="${SCRIPT_DIR}/worlds/${WORLD}"
+    gzserver --verbose $SDFFILE &
+    GZID=$!
+
+    # Wait until gazebo is up and running
+    sleep 8
+
+    SOCAT_ARG_2="udp:localhost:$GZSITL_UDP_PORT"
+    if (("$AUTOPILOT" == "$AP_PX4")); then
+        SOCAT_ARG_1="udp-listen:$PX4_UDP_PORT_1"
+    elif (("$AUTOPILOT" == "$AP_APM")); then
+        SOCAT_ARG_1="tcp:localhost:$APM_TCP_PORT_1"
+    fi
+
+    # Bidirectional bridge between autopilot and gazebo-sitl
+    socat $SOCAT_ARG_1 $SOCAT_ARG_2 &
+    GZSITL_SOCATID=$!
+
+    # Wait for gzsitl-sitl connection to be stabilished
+    sleep 2
+}
+
+run_coav_control() {
+    # Run the collision avoidance gcs
+    ../build/tools/coav-control/coav-control "$@" &
+    COAVID=$!
+
+    # Wait until gcs is up and running
+    sleep 4
+
+    SOCAT_ARG_2="udp:localhost:$COAV_GCS_UDP_PORT"
+    if (("$AUTOPILOT" == "$AP_PX4")); then
+        SOCAT_ARG_1="udp-listen:$PX4_UDP_PORT_2"
+    elif (("$AUTOPILOT" == "$AP_APM")); then
+        SOCAT_ARG_1="tcp:localhost:$APM_TCP_PORT_2"
+    fi
+
+    # Bidirectional bridge between the autopilot and the coav_gcs
+    socat $SOCAT_ARG_1 $SOCAT_ARG_2 &
+    COAV_SOCATID=$!
+}
+
+
+simulate () {
+    run_autopilot
+    run_gazebo
+    run_coav_control "$@"
+
+    # wait forever
+    cat
+}
+
+silentkill () {
+    if [ ! -z $2 ]; then
+        kill $2 $1 > /dev/null 2>&1 || true
+    else
+        kill -KILL $1 > /dev/null 2>&1 && wait $1 2> /dev/null || true
+    fi
+}
+
+cleanup_and_exit () {
+    silentkill $GZSITL_SOCATID # Kill gzsitl socat
+    silentkill $COAV_SOCATID # Kill coav_gcs socat
+    silentkill $SITLID # Kill sitl
+    silentkill $GZID -INT && sleep 3 # Wait gzserver to save the log
+    silentkill $GZID  # Kill gzserver
+    silentkill $COAVID # Kill coav-control sitl
+    exit 0
+}
+
+test_dep () {
+    command -v $1 > /dev/null 2>&1 || { echo >&2 "Error: command '$1' not found"; exit 1; }
+}
+
+check_deps () {
+    test_dep gz
```

Comment:
It is true that both should be present, but you are right, it is safer to check for gzserver directly.